core/src/test/java/kafka/test/api/ProducerRebootstrapTest.java

```diff
@@ -0,0 +1,121 @@
+
+        return Stream.of(false, true)
+                .map(ProducerRebootstrapTest::getRebootstrapConfig)
+                .map(rebootstrapProperties -> ProducerRebootstrapTest.buildConfig(serverProperties, rebootstrapProperties))
+                .toList();
+    }
+
+    static Map<String, String> getRebootstrapConfig(boolean useRebootstrapTriggerMs) {
+        Map<String, String> properties = new HashMap<>();
+        if (useRebootstrapTriggerMs) {
+            properties.put("metadata.recovery.rebootstrap.trigger.ms", "5000");
+        } else {
+            properties.put("metadata.recovery.rebootstrap.trigger.ms", "3600000");
+            properties.put("socket.connection.setup.timeout.ms", "5000");
+            properties.put("socket.connection.setup.timeout.max.ms", "5000");
+            properties.put("reconnect.backoff.ms", "1000");
+            properties.put("reconnect.backoff.max.ms", "1000");
+        }
+        properties.put("metadata.recovery.strategy", "rebootstrap");
+        properties.putIfAbsent(ProducerConfig.ACKS_CONFIG, "-1");
+        return properties;
+    }
+
+    static ClusterConfig buildConfig(Map<String, String> serverProperties, Map<String, String> rebootstrapProperties) {
+        return ClusterConfig.defaultBuilder()
+                .setTypes(Set.of(Type.KRAFT))
+                .setBrokers(BROKER_COUNT)
+                .setProducerProperties(rebootstrapProperties)
+                .setServerProperties(serverProperties).build();
+    }
+
+    @ClusterTemplate(value = "generator")
+    public void testRebootstrap(ClusterInstance clusterInstance) throws ExecutionException, InterruptedException {
+        String topic = "topic";
+        try (Admin admin = clusterInstance.admin()) {
+            admin.createTopics(Collections.singletonList(new NewTopic(topic, BROKER_COUNT, (short) 2))).all().get();
+        }
+
+        int part = 0;
+        KafkaBroker server0 = clusterInstance.brokers().get(0);
+        KafkaBroker server1 = clusterInstance.brokers().get(1);
+
+        // It's ok to shut the leader down, cause the reelection is small enough to the producer timeout.
+        server1.shutdown();
+        server1.awaitShutdown();
+
+        try (var producer = clusterInstance.producer()) {
+            // Only the server 0 is available for the producer during the bootstrap.
+            RecordMetadata recordMetadata0 = producer.send(new ProducerRecord<>(topic, part, "key 1".getBytes(), "value 1".getBytes())).get();
+            assertEquals(0, recordMetadata0.offset());
+
+            server0.shutdown();
+            server0.awaitShutdown();
+            server1.startup();
+
+            // Current server 0 is offline.
+            // However, the server 1 from the bootstrap list is online.
+            // Should be able to produce records.
+            RecordMetadata recordMetadata1 = producer.send(new ProducerRecord<>(topic, part, "key 1".getBytes(), "value 1".getBytes())).get();
```

Comment:
ditto: Variables should be declared using var to maintain consistency throughout the file.